storage-local/src/main/java/io/kestra/storage/local/LocalStorage.java

```diff
@@ -79,9 +73,15 @@ public Long lastModifiedTime(URI uri) throws IOException {
         return lastModifiedTime.toMillis();
     }
 
+    @Override
+    public BasicFileAttributes getAttributes(URI uri) throws IOException {
+        return Files.readAttributes(getPath(uri), BasicFileAttributes.class);
+    }
+
     @Override
     public URI put(URI uri, InputStream data) throws IOException {
-        this.createDirectory(uri);
+        Path path = Paths.get(config.getBasePath().toAbsolutePath().toString(), uri.getPath());
```

Comment:
```suggestion
        Path path = Paths.get(uri.getPath());
```
I think if you don't do that before getting parent you will end up with the config.getBasePath() append twice

Reply:
Beware that getParent() could be null I think if the uri is a single file with no nesting